src/common/components/markdownViewer/syncScroll/synchronizeScroll.ts

```diff
@@ -0,0 +1,113 @@
+      }
+      candidate = current;
+    }
+  }
+  return { exactMatch: false, candidate };
+};
+
+// Retrieve the candidate(s) elements that are located in a certain pixel offset
+// position from an input list of rendered elements.
+const getRenderedElementsForPixelOffset = (elementList: NodeListOf<Element>, targetOffset: number): CandidateSearch => {
+    let candidate: Candidate = null;
+    let lineFraction = 0;
+    for (const element of elementList) {
+      const slAttr = element.getAttribute(SOURCE_LINE_ATTRIBUTE);
+      if (isIntegerProperty(slAttr)) {
+        const boundingRect = element.getBoundingClientRect();
+        const current: Candidate = { lineNum: Number(slAttr), element };
+        if (boundingRect.top > targetOffset) {
+          if (candidate && lineFraction < 1) {
+            candidate.lineNum += lineFraction;
+            return { exactMatch: true, candidate };
+          }
+          return { exactMatch: false, candidate, nextCandidate: current };
+        }
+        lineFraction = (targetOffset - boundingRect.top) / boundingRect.height;
+        candidate = current;
+      }
+    }
+    return { exactMatch: false, candidate };
+};
+
+// Fine tune this constant to place our reference 'marker' at a certain
+// distance of the component top border.
+const PADDING_OFFSET = 0;
+
+// For a given list of rendered elements (each element coming from a source line) and
+// a target line, returns the pixel offser for the viewport to show the element matching
+// the target line.
+const getPixelOffsetForSourceLine = (elementList: NodeListOf<Element>, targetLine: number): number => {
+  const cs = getRenderedElementsForSourceLine(elementList, targetLine);
+  if (cs.candidate && cs.candidate.element) {
+    const inBetween = cs.nextCandidate ?
+      (targetLine - cs.candidate.lineNum)
+      * (cs.nextCandidate.element.getBoundingClientRect().top - cs.candidate.element.getBoundingClientRect().top)
+      / (cs.nextCandidate.lineNum - cs.candidate.lineNum)
+      : 0;
+    return cs.candidate.element.getBoundingClientRect().top + inBetween - PADDING_OFFSET;
+  }
+  return 0;
+};
+
+// For a given list of rendered elements (each element coming from a source line) and
+// a target offset, returns the line number of the element located in that target offset.
+const getSourceLineForPixelOffset = (elementList: NodeListOf<Element>, targetOffset: number): number => {
+  const adjustedOffset = targetOffset + PADDING_OFFSET;
+  const cs = getRenderedElementsForPixelOffset(elementList, adjustedOffset);
+  if (cs.candidate) {
+    const inBetween = cs.nextCandidate ?
+    (adjustedOffset - cs.candidate.element.getBoundingClientRect().top)
+    / (cs.nextCandidate.element.getBoundingClientRect().top - cs.candidate.element.getBoundingClientRect().top)
+    * (cs.nextCandidate.lineNum - cs.candidate.lineNum)
+    : 0;
+    return cs.candidate.lineNum + inBetween;
+  }
+  return 0;
+};
```

Comment:
Same here we can encapsulate this in submethods and make the code more readble

Reply:
Check refactor as commented above. [Link to commit](https://github.com/Lemoncode/LeanMood/pull/168/commits/d9cd50cd321fb25e7f3fa0159abe6bb07c3ef661).